src/app/organizations/manage/reset-password.component.ts

```diff
@@ -0,0 +1,196 @@
+
+    async ngOnInit() {
+        // Get Enforced Policy Options
+        this.enforcedPolicyOptions = await this.policyService.getMasterPasswordPolicyOptions();
+    }
+
+    get loggedOutWarningName() {
+        return this.name != null ? this.name : this.i18nService.t('thisUser');
+    }
+
+    getPasswordScoreAlertDisplay() {
+        if (this.enforcedPolicyOptions == null) {
+            return '';
+        }
+
+        let str: string;
+        switch (this.enforcedPolicyOptions.minComplexity) {
+            case 4:
+                str = this.i18nService.t('strong');
+                break;
+            case 3:
+                str = this.i18nService.t('good');
+                break;
+            default:
+                str = this.i18nService.t('weak');
+                break;
+        }
+        return str + ' (' + this.enforcedPolicyOptions.minComplexity + ')';
+    }
+
+    async generatePassword() {
+        const options = (await this.passwordGenerationService.getOptions())[0];
+        this.newPassword = await this.passwordGenerationService.generatePassword(options);
+        this.updatePasswordStrength();
+    }
+
+    togglePassword() {
+        this.showPassword = !this.showPassword;
+        document.getElementById('newPassword').focus();
+    }
+
+    copy(value: string) {
+        if (value == null) {
+            return;
+        }
+
+        this.platformUtilsService.copyToClipboard(value, { window: window });
+        this.platformUtilsService.showToast('info', null,
+            this.i18nService.t('valueCopied', this.i18nService.t('password')));
+    }
+
+    async submit() {
+        // Validation
+        if (this.newPassword == null || this.newPassword === '') {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPassRequired'));
+            return false;
+        }
+
+        if (this.newPassword.length < 8) {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPassLength'));
+            return false;
+        }
+
+        if (this.enforcedPolicyOptions != null &&
+            !this.policyService.evaluateMasterPassword(this.masterPasswordScore, this.newPassword,
+                this.enforcedPolicyOptions)) {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPasswordPolicyRequirementsNotMet'));
+            return;
+        }
+
+        if (this.masterPasswordScore < 3) {
+            const result = await this.platformUtilsService.showDialog(this.i18nService.t('weakMasterPasswordDesc'),
+                this.i18nService.t('weakMasterPassword'), this.i18nService.t('yes'), this.i18nService.t('no'),
+                'warning');
+            if (!result) {
+                return false;
+            }
+        }
```

Comment:
Same here, would be nice if we could extract this and ensure we don't forget any rules in the future.